gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/DefaultIdentityProviderServiceImpl.java

```diff
@@ -0,0 +1,268 @@
+                }
+            }
+
+            configMap.put("host", jdbcHost());
+            configMap.put("port", jdbcPort());
+            configMap.put("protocol", jdbcDriver());
+            configMap.put("database", jdbcDatabase());
+            // dash are forbidden in table name, replace them in domainName by underscore
+            configMap.put("usersTable", "idp_users_" + tableSuffix);
+            configMap.put("user", jdbcUser());
+            configMap.put(PASSWORD, (jdbcPassword() == null ? null : jdbcPassword()));
+            configMap.put("autoProvisioning", idpProvisioning());
+            configMap.put("selectUserByUsernameQuery", "SELECT * FROM idp_users_" + tableSuffix + " WHERE username = %s");
+            configMap.put("selectUserByEmailQuery", "SELECT * FROM idp_users_" + tableSuffix + " WHERE email = %s");
+            configMap.put("identifierAttribute", "id");
+            configMap.put("usernameAttribute", "username");
+            configMap.put("passwordAttribute", PASSWORD);
+            configMap.put("passwordEncoder", encoder);
+            updatePasswordEncoderOptions(configMap, encoder);
+        }
+        return configMap;
+    }
+
+    public String addOptionsToURI(String mongoUri) {
+        Integer connectTimeout = environment.getProperty("management.mongodb.connectTimeout", Integer.class, 1000);
+        Integer socketTimeout = environment.getProperty("management.mongodb.socketTimeout", Integer.class, 1000);
+        Integer maxConnectionIdleTime = environment.getProperty("management.mongodb.maxConnectionIdleTime", Integer.class);
+        Integer heartbeatFrequency = environment.getProperty("management.mongodb.heartbeatFrequency", Integer.class);
+        Boolean sslEnabled = environment.getProperty("management.mongodb.sslEnabled", Boolean.class);
+        String authSource = environment.getProperty("management.mongodb.authSource", String.class);
+
+        mongoUri += mongoUri.endsWith("/") ? "" : "/";
+        mongoUri += "?connectTimeoutMS=" + connectTimeout + "&socketTimeoutMS=" + socketTimeout;
+        if (authSource != null) {
+            mongoUri += "&authSource=" + authSource;
+        }
+        if (maxConnectionIdleTime != null) {
+            mongoUri += "&maxIdleTimeMS=" + maxConnectionIdleTime;
+        }
+        if (heartbeatFrequency != null) {
+            mongoUri += "&heartbeatFrequencyMS=" + heartbeatFrequency;
+        }
+        if (sslEnabled != null) {
+            mongoUri += "&ssl=" + sslEnabled;
+        }
+
+        return mongoUri;
+    }
+
+
+    private String createProviderConfig(String referenceId, NewIdentityProvider identityProvider) {
+        final Map<String, Object> configMap = createProviderConfiguration(referenceId, identityProvider);
+        try {
+            return new ObjectMapper().writeValueAsString(configMap);
+        } catch (JsonProcessingException e) {
```

Comment:
ObjectMapper could be a dependecy on the DefaultIdentityProviderServiceImpl

Reply:
There is no Bean created with ObjectMapper in this package.

Reply:
ObjectMapper creation can be expensive (it loads modules etc). How about adding it as final mapper = new ObjectMapper at class level?